### PR TITLE
fix(clarinet): copy substring text to avoid reference to original chunk

### LINKED ISSUE #154: oboe.drop doesn't free memory in my case
Hi all,

I came across oboe recently and my first impression was pretty much positive! I really like how smooth it works. Then, however, I used bigger datasets and my browser began to crash.
Obviously, there is a memory problem with oboe.

This is how my code looks right now:

```
var myOboeObject = oboe({
            url: '/someURL/',
            method: 'POST',
            headers: {
                authorization: { /* .... */ }
            },
            body: { /* .... */ }
        });

        var myfooArray = [];
        myOboeObject.on('node','{some_property_which_is_everwhere_there}',function(node) {
            myfooArray.push(node);
            return oboe.drop;
        });
```

The problem seems to be that `oboe.drop` doesn't work as expected for me. It seems to work if there I don't do anything with the `node` parameter. Then the memory gets freed immediately and there is no problem in the browser.

However, as soon the `node` parameter is used somehow (even a `console.log` or just using one property of it or even copying it with `angular.copy`) the memory is not freed and eventually, the browser will crash.

See image attached:

oboe_mem1

Am I doing something wrong? Or is this a problem of oboe?
I'm a little bit desparate right now because I cannot have browser crashes, even very much lower performance is better than crashes because of too much memory used.

### PR DESCRIPTION
**Pull request by Aigeec**
Looking to fix #154 

Need to look at tests for it. As is a V8 issue testing it in Node might be an option.

@willkara You might be able to use the dist files I've pushed up to do some preliminary testing.

### CODE REVIEW

**Comment by JuanCaicedo** [comment]
We could probably do a release after this is merged 😀

**Comment by Aigeec** [comment]
I can prove this works with Chrome Version 65.0.3325.162 (Official Build) (64-bit) just not sure how to do it programmatically.

There is another possible leak in that the responseText seems to hang around on the heap for the life of the oboe instance. I would have though that this should be destroyed once we are finished processing or even as we are processing.

**Comment by Aigeec** [comment]
Have a test that works with headless chrome. Should be able to run it on Travis with puppeteer.

It think there are some other leaks around the xhr. The responseText should be collected one the request is complete. Will look at that.

**Comment by Aigeec** [comment]
So if you look at build [#353](https://travis-ci.org/jimhigson/oboe.js/builds/355169181#) (before the fix is applied) you'll see that end heap is 585MB. After the fix is applied in build [#357](https://travis-ci.org/jimhigson/oboe.js/builds/355172660) the end heap is 63MB.

Actually doing this in chrome and taking a snap shot shows a heap of about 15MB. This is mostly accounted for by the array that contains each of the nodes and the responseText being held onto by the xhr.

There will always be an issue for people with neverending requests as there is no way for use to clear the processed responseText. So as long as they are adding to it it will not be garbage collected.

Using the Fetch API may solve that.

File: src/streamingHttp.browser.js [hunk 1]
```diff
@@ -72,26 +72,26 @@ function streamingHttp (oboeBus, xhr, method, url, data, headers, withCredential
...
-      // be read so has to be protected with try/catch and tried again on
-      // the next readyState if it fails
-      try {
-        stillToSendStartEvent && oboeBus(HTTP_START).emit(
-          xhr.status,
-          parseResponseHeaders(xhr.getAllResponseHeaders()))
-        stillToSendStartEvent = false
-      } catch (e) { /* do nothing, will try again on next readyState */ }
-    }
+  function sendStartIfNotAlready (xhr) {
```
Comment (JuanCaicedo) [inline src/streamingHttp.browser.js]: Nice change! 👍

**Pull request merged**